spec/lib/personas/tools/researcher_spec.rb

```diff
@@ -21,6 +21,54 @@
 
   before { SiteSetting.ai_bot_enabled = true }
 
+  it "uses custom researcher_llm and applies token limits correctly" do
+    # Create a second LLM model to test the researcher_llm option
+    secondary_llm_model = Fabricate(:llm_model, name: "secondary_model")
+
+    # Create test content with long text to test token truncation
+    topic = Fabricate(:topic, category: category, tags: [tag_research])
+    long_content = "zz " * 100 # This will exceed our token limit
```

Comment:
`100` is abit of a magic number here. I wonder if we can generate the long content here based on a constant so that updating the constant in the future doesn't break this test.

Reply:
Actually is this limit based on https://github.com/discourse/discourse-ai/pull/1401/files#diff-a4d7be1d2a5b82a80a1a4693975f7ac231a577ced032e4d941f332292fb56339R47? If so, then I think this is fine.